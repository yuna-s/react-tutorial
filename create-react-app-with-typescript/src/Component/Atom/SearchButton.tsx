import React, { ReactChildren, ComponentProps } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { makeStyles, createStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(theme => createStyles({
  ButtonControl: {
    zIndex: theme.zIndex.drawer + 1,
    marginTop: 20,
    marginBottom: 20,
  }
}))


export default function SearchButton(props: ButtonProps) {
  const classes = useStyles();
  return (
    <Button variant="contained" color="primary" href="#contained-buttons" fullWidth className={classes.ButtonControl}>
      {props.label}
    </ Button>
  );
}


export interface ButtonProps {
  label: String
};