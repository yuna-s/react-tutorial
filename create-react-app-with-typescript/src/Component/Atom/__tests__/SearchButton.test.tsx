import React from 'react';
import Button from '@material-ui/core/Button';

import SearchButton from '../SearchButton';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
configure({ adapter: new Adapter() });


describe('SearchButton', (): void => {
  const container = mount(<SearchButton label='test' />);

  test('ボタンが一つか.', () => {
    expect(container.find(Button).length).toBe(1);
  });

  test('ラベルpropsに与えた文字が表示されるか.', () => {
    expect(container.find(Button).html()).toContain('test');
  });
})