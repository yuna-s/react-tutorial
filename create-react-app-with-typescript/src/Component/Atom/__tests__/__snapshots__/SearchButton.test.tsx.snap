// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should make normal button. 1`] = `ShallowWrapper {}`;
